crates/uv-toolchain/src/interpreter.rs

```diff
@@ -643,6 +645,10 @@ impl InterpreterInfo {
             format!("{}.msgpack", digest(&uv_fs::absolutize_path(executable)?)),
         );
 
+        if !executable.try_exists()? {
+            return Err(Error::NotFound(executable.to_path_buf()));
```

Comment:
Can we instead match on `uv_fs::canonicalize_executable(executable)`, and just do this in the `NotFound` branch of the error? Otherwise, we're doing another fs operation that we can avoid.

Reply:
Sure, I was worried that wasn't future proof but it _is_ faster...